code/datums/components/decals/blood.dm

```diff
@@ -0,0 +1,25 @@
+
+/datum/component/decal/blood
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+/datum/component/decal/blood/Initialize(_icon, _icon_state, _dir, _cleanable=CLEAN_STRENGTH_BLOOD, _color, _layer=ABOVE_OBJ_LAYER)
+	if(isitem(parent))
+		return ..()
+	else
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/decal/blood/generate_appearance(_icon, _icon_state, _dir, _layer, _color)
+	var/obj/item/I = parent
+	if(initial(I.icon) && initial(I.icon_state))
+		//try to find a pre-processed blood-splatter. otherwise, make a new one
+		var/index = I.blood_splatter_index()
+		var/icon/blood_splatter_icon = GLOB.blood_splatter_icons[index]
+		if(!blood_splatter_icon)
+			blood_splatter_icon = icon(initial(I.icon), initial(I.icon_state), , 1)		//we only want to apply blood-splatters to the initial icon_state for each object
+			blood_splatter_icon.Blend("#fff", ICON_ADD) 			//fills the icon_state with white (except where it's transparent)
+			blood_splatter_icon.Blend(icon('icons/effects/blood.dmi', "itemblood"), ICON_MULTIPLY) //adds blood and the remaining white areas become transparant
+			GLOB.blood_splatter_icons[index] = blood_splatter_icon

```

Comment:
what's the point of making blood a decal when you're going to keep one of the worst examples of old shitcode alive in it

Reply:
how else do I keep track of generated icons so they aren't regenerated each time?

Reply:
You do realize this was made before overlays were a thing right?

Reply:
You do realise this is blending itself onto the icon, such that blood only appears on the parts of the icon that aren't transparent, IE: blood only splatters on the original sprite itself, not blood floating in mid air.

Something overlays can't do.

Reply:
yes but it's creating something that should be applied as an overlay

Reply:
which it does, later on, which is now handled in decal component's pic var.
(previously it directly add_overlay()'d) itself.)

Reply:
new maintainers